packages/grafana-llm-app/pkg/plugin/app.go

```diff
@@ -71,7 +72,12 @@ func NewApp(ctx context.Context, appSettings backend.AppInstanceSettings) (insta
 	app.CallResourceHandler = httpadapter.New(mux)
 
 	// Getting the service account token that has been shared with the plugin
-	app.saToken = os.Getenv("GF_PLUGIN_APP_CLIENT_SECRET")
+	cfg := backend.GrafanaConfigFromContext(ctx)
+	app.saToken, err = cfg.PluginAppClientSecret()
+	if err != nil {
+		log.DefaultLogger.Error("Error getting service account token")
+		return nil, fmt.Errorf("error getting service account token: %w", err)
```

Comment:
This is a breaking change if your Grafana instance doesn't have the `externalServiceAccounts` feature toggle turned on. Think we should log a warning and not return an error here. Or we could also first check if that feature is toggled on before trying to get the SA token this way

Reply:
Hmm, that's a good point. Will change to warning.